2-Draft-Accepted/RFCxxxx-DSC-for-PowerShell7.md

```diff
@@ -0,0 +1,117 @@
+
+Custom agents can simply use the existing PowerShell API (or call out to `pwsh`) to directly invoke DSC resources.
+
+You can even simply write a PowerShell script as your agent that runs off TaskScheduler or cron, for example.
+
+Along with the lack of a default agent, there is no pre-defined push or pull model for deploying DSC.
+
+### Cross platform first
+
+DSC has supported both Windows and Linux, however, the [Linux support](https://docs.microsoft.com/powershell/scripting/dsc/getting-started/lnxGettingStarted?view=powershell-7)
+was dependent on [OMI](https://github.com/Microsoft/omi) and did not support authoring Linux DSC resources in PowerShell script making development
+inconsistent and much more difficult.
+
+DSC for PowerShell 7 will be cross platform having consistent capabilities and user experience on all platforms.
+
+### Moving away from WMI/OMI/MOF
+
+DSC on Windows was originally designed relying on [WMI](https://docs.microsoft.com/windows/win32/wmisdk/wmi-start-page) as a host on Windows,
+OMI as a host on Linux, and [MOF](https://docs.microsoft.com/windows/win32/wmisdk/managed-object-format--mof-) for configuratio and schema.
+
+MOF as a format is not something most developers are familiar with and not much tooling support.
+However, developers of DSC resources written in PowerShell script still needed to learn MOF and also keep their MOF schema
+in sync with their script implementation.
+
+Today, when you execute a [DSC configuration script](https://docs.microsoft.com/powershell/scripting/dsc/configurations/configurations?view=powershell-7),
+the output is a MOF configuration file.
+In general, users do not need to read this MOF file as it's intended to be used by machines.
+However, MOF parsing creates a dependency on WMI/OMI.
+Instead, we will move towards a JSON configuration file format.
+
+Since there is no default agent to understand the JSON configuration file, it will be up to users or 3rd party agents to understand the configuration file.
+
+<TODO> example JSON configuratoin </TODO>
+
+### No more MOF schema and script resources
+
+Although DSC resources can be written as WMI/OMI providers, this requires writing C/C++ code and compiling for specific Linux distros and processor architectures.
+Writing PowerShell script DSC resources currently still requires a matching schema file making it more complex.
+Instead, the current plan is to [only support PowerShell class based DSC resources](https://github.com/PowerShell/PowerShell/issues/13731) in DSC for PowerShell 7.
+The reasoning and discussion is in the issue linked above.
+
+### Open Source
+
+The current [PSDesiredStateConfiguration](https://docs.microsoft.com/powershell/module/psdesiredstateconfiguration/?view=powershell-7) module is not Open Source.
+The intent is to clean up the code and prepare the GitHub repository to be made public as an Open Source project.
```

Comment:
You say clean up the code, but you don't mention touching the DSL... Intentional omission?